### ISSUE TITLE
eslint configuration for jsx

### ISSUE THREAD

**Issue by matt-e-king** [eslint/eslint#2277]
Hello there! In the documentation on http://eslint.org/docs/user-guide/configuring, under Specifying Environments, it states:

```es6``` - enable all ECMAScript 6 features except for modules.

So as this line states, I only included ```modules: true``` - so initially my .eslintrc file looked like:
```
{
  "env": {
    "browser": true,
    "node": true,
    "es6": true
  },
  "ecmaFeatures": {
    "modules": true
  }
}
```
However, I was getting ```Unexpected token < ``` linting errors on my JSX. I resolved this by adding ```jsx: true``` rule to ```ecmaFeatures``` 
```
{
  "env": {
    "browser": true,
    "node": true,
    "es6": true
  },
  "ecmaFeatures": {
    "modules": true,
    "jsx": true
  }
}
```
And now everything lints as expected. Does the documentation need to be updated? Or is there something else I am overlooking? Thanks!

**Comment by ilyavolodin** [comment]
Well, JSX isn't part of ES6, so documentation seems to be correct.

**Comment by matt-e-king** [comment]
Forgive me, I keep seeing React and ES6 as interchangeable since I am learning them at the same time.

But I guess it raises the question that since JSX isn't part of the ECMAScript spec, should it even belong as a property to ```ecmaFeatures``` in the config file?

**Comment by nzakas** [comment]
This comes from Espree, where all extensions to ECMAScript are configured. This does not relate in any way to something being in the spec or not.
